### ISSUE TITLE
NC 16.0.1: Clicking on the contacts menu doesn't show the contacts image anymore

### ISSUE THREAD

**Issue by HigH-HawK** [nextcloud/server#15636]
### Steps to reproduce
1. Click on the contacts menu in the upper right hand corner and wait for the menu to open
2. The menu appears with the contacts but their image are not being shown anymore

### Expected behaviour
The contacts menu should open, show the contacts with their images

### Actual behaviour
See steps to reproduce

### Server configuration

**Operating system**: CentOS 7

**Web server:** Apache/2.4.6

**Database:** 15.1 Distrib 10.3.15-MariaDB

**PHP version:** PHP 7.2.17

**Nextcloud version:** 16.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 15.0.7

**Where did you install Nextcloud from:** 

**Signing status:** 
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.1
  - admin_audit: 1.6.0
  - bruteforcesettings: 1.3.0
  - cloud_federation_api: 0.2.0
  - dav: 1.9.2
  - encryption: 2.4.0
  - federatedfilesharing: 1.6.0
  - files: 1.11.0
  - files_markdown: 2.0.6
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.13.0
  - files_sharing: 1.8.0
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - files_videoplayer: 1.5.0
  - gallery: 18.3.0
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - notes: 2.6.0
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - twofactor_totp: 2.1.2
  - updatenotification: 1.6.0
  - user_ldap: 1.6.0
  - viewer: 1.0.0
  - workflowengine: 1.6.0
Disabled:
  - comments
  - federation
  - files_external
  - firstrunwizard
  - nextcloud_announcements
  - password_policy
  - support
  - survey_client
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
<?php
$CONFIG = array (
  'instanceid' => '***REMOVED SENSITIVE VALUE***',
  'passwordsalt' => '***REMOVED SENSITIVE VALUE***',
  'secret' => '***REMOVED SENSITIVE VALUE***',
  'trusted_domains' =>
  array (
    0 => 'Host1',
    1 => 'Host2',
    2 => 'IP',
  ),
  'datadirectory' => '***REMOVED SENSITIVE VALUE***',
  'dbtype' => 'mysql',
  'version' => '16.0.1.1',
  'overwrite.cli.url' => 'https://***REMOVED SENSITIVE VALUE***',
  'proxy' => '***REMOVED SENSITIVE VALUE***',
  'dbname' => '***REMOVED SENSITIVE VALUE***',
  'dbhost' => '***REMOVED SENSITIVE VALUE***',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => '***REMOVED SENSITIVE VALUE***',
  'dbpassword' => '***REMOVED SENSITIVE VALUE***',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'ldapProviderFactory' => 'OCA\\User_LDAP\\LDAPProviderFactory',
  'mail_smtpmode' => 'smtp',
  'mail_smtpsecure' => 'ssl',
  'mail_from_address' => '***REMOVED SENSITIVE VALUE***',
  'mail_domain' => '***REMOVED SENSITIVE VALUE***',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => '***REMOVED SENSITIVE VALUE***',
  'mail_smtpport' => '465',
  'mail_smtpname' => '***REMOVED SENSITIVE VALUE***',
  'mail_smtppassword' => '***REMOVED SENSITIVE VALUE***',
  'maintenance' => false,
  'log_rotate_size' => 10485760,
  'theme' => '',
  'loglevel' => 2,
  'lost_password_link' => 'disabled',
  'updater.release.channel' => 'stable',
  'mysql.utf8mb4' => true,
);
```
</details>

**Are you using encryption:** yes

### Client configuration
**Browser:** Chrome 74.0.3729.157 and Firefox 66.0.5

The website is being opened via https:// but as soon as I open the contacts menu, the browser tells me that the connection is not secure anymore. After using the web developer tools from the browser I found out that the images of the contacts are being linked as http:// and not https://

As soon as I change the protocol from http:// to https:// in the web developer tools, the image appears. This is due to the browser blocking http:// as not secure.

Screenshot of how it looks:
![nc_contacts](https://user-images.githubusercontent.com/33863449/58034700-79dd5480-7b27-11e9-9ce6-8791a352bf03.png)

Problem seems to be in the `remote.php`

**Comment by HigH-HawK** [comment]
I was able to locate the template for this menu and noticed that it uses a variable `contact.avatar` but I cannot find this variable anywhere else, to get an idea of how the actual link to the avatar is being build.

There must be a way to either set the protocol to `https://` or just replace it with `://` (last would be ideal as it would just orientate itself by the protocol being used upon opening the website.

**Comment by HigH-HawK** [comment]
This is still an issue, would it be possible to fix this in one of the future versions?

**Comment by j-ed** [comment]
I couldn't replicate your problem on 16.0.1 nor in 16.0.2 and I haven't read that any other person has reported this kind of problem. So most likely it is a problem which only appears on your installation.
Maybe it is a proxy related problem so that the protocol is not correctly set?!

**Comment by HigH-HawK** [comment]
Strange is, it only does that in the addressbook i.e. if I open the contacts menu. My own avatar and the avatar of a user sharing stuff with me, are being displayed fine.

**Comment by HigH-HawK** [comment]
Solved with the link in #16370 from @kesselb
